packages/jest-snapshot/src/__tests__/utils.test.ts

```diff
@@ -72,9 +72,9 @@ test('saveSnapshotFile() works with \r', () => {
 
 test('getSnapshotData() throws when no snapshot version', () => {
   const filename = path.join(__dirname, 'old-snapshot.snap');
-  jest
-    .mocked(fs.readFileSync)
-    .mockImplementation(() => 'exports[`myKey`] = `<div>\n</div>`;\n');
+  (fs.readFileSync as unknown as jest.Mocked<() => string>).mockImplementation(
+    () => 'exports[`myKey`] = `<div>\n</div>`;\n',
+  );
```

Comment:
Oh.. `jest.mocked()` just does not work with the return type of `fs.readFileSync`. It cannot be just `string` or just `Buffer`, it is required to be `string | Buffer`.

Reply:
you can add a `typeof` check to the argument

Reply:
Solved it. Just works like this:

```ts
jest
  .mocked(fs.readFileSync)
  .mockReturnValue('exports[`myKey`] = `<div>\n</div>`;\n');
```

Reply:
sorta weird `mockReturnValue` allows you to ignore return type?

Reply:
Function typings are coming from overloads. So implementation is expecting something what covers all overloads. Seemed like `ReturnType` skipped this idea and allowed simply to have `string | Buffer`.